compose/project.py

```diff
@@ -302,15 +303,17 @@ def up(self,
         if self.use_networking:
             self.ensure_network_exists()
 
-        return [
-            container
-            for service in services
-            for container in service.execute_convergence_plan(
+        containers = []
+        for service in services:
+            converged_results = service.execute_convergence_plan(
                 plans[service.name],
                 do_build=do_build,
-                timeout=timeout
+                timeout=timeout,
+                detached=detached
             )
-        ]
+            containers += converged_results
+
+        return containers
```

Comment:
This doesn't need to change, right? It's just an added parameter.  I think the compression version is more clear.